src/control/server/server_utils.go

```diff
@@ -208,20 +208,23 @@ func getFabricNetDevClass(cfg *config.Server, fis *hardware.FabricInterfaceSet)
 
 // Detect if any engines share numa nodes and if that's the case, allocate only on the shared numa
 // node and notify user.
-func getEngineNUMANodes(log logging.Logger, engineCfgs []*engine.Config) []string {
-	nodeMap := make(map[string]bool)
-	nodes := make([]string, 0, len(engineCfgs))
+func getEngineNUMANodes(log logging.Logger, engineCfgs []*engine.Config) []int {
+	nodeMap := make(map[int]int)
 	for _, ec := range engineCfgs {
-		nn := fmt.Sprintf("%d", ec.Storage.NumaNodeIndex)
-		if nodeMap[nn] {
-			log.Noticef("Multiple engines assigned to NUMA node %s, "+
-				"allocating all hugepages on this node.", nn)
-			nodes = []string{nn}
-			break
+		nodeMap[int(ec.Storage.NumaNodeIndex)] += 1
+	}
+
+	var lastCount int
+	nodes := make([]int, 0, len(engineCfgs))
+	for k, v := range nodeMap {
+		if lastCount != 0 && v != lastCount {
+			log.Errorf("different number of engines assigned to each NUMA node (%+v),"+
```

Comment:
Seems like more of a Notice to me

Reply:
It is an error condition, having a different number of engines running on each NUMA will cause problems so I think it should be raised as an error in the log but for extraneous circumstances we want to allow the server to start in this case. As an aside, it was previously a Notice but wasn't being printed for some reason when the log mask was set to DEBUG.

Reply:
I changed this to return an error.